src/main/java/org/roaringbitmap/RunContainer.java

```diff
@@ -942,6 +932,40 @@ public boolean equals(Object o) {
     return false;
   }
 
+  private boolean equals(RunContainer runContainer) {
+    if (runContainer.nbrruns != this.nbrruns) {
+      return false;
+    }
+    for (int i = 0; i < nbrruns; ++i) {
+      if (getValue(i) != runContainer.getValue(i)) {
+        return false;
+      }
+      if (getLength(i) != runContainer.getLength(i)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  private boolean equals(ArrayContainer arrayContainer) {
+    if (getCardinality() != arrayContainer.getCardinality()) {
```

Comment:
I ran a few scenarios, the difference in execution speed is hardly noticeable, but (as you'd expect) you can start to see it when you create lots of short runs. _Probabalistically_ the more short runs you have in the range [0,4096) the less likely you are to have an array container with exactly the same bits set, so I doubt you would ever get past this first check when the number of runs is large enough to make a difference. In any case, I changed it, and added a benchmark for many small runs with cardinality less than 4096.